scripts/tracecompass-server-download.ts

```diff
@@ -0,0 +1,101 @@
+/********************************************************************************
+ * Copyright (C) 2022 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import * as cliProgress from "cli-progress";
+import * as fs from "fs-extra";
+import * as request from "request";
+import * as tar from "tar-fs";
+import gunzip from "gunzip-maybe";
+
+const LOCAL_DOWNLOAD_TARGET_PATH = "./tracecompass-server/";
+const LOCAL_DOWNLOAD_TARGET_FILENAME = "trace-server.tar.gz";
+const LOCAL_BINARY_PATH = `${LOCAL_DOWNLOAD_TARGET_PATH}trace-compass-server/tracecompass-server`;
+const DOWNLOAD_URL =
+  "https://download.eclipse.org/tracecompass.incubator/trace-server/rcp/trace-compass-server-latest-linux.gtk.x86_64.tar.gz";
+
+execute();
+
+async function execute(): Promise<void> {
+  try {
+    const archivePath = `${LOCAL_DOWNLOAD_TARGET_PATH}${LOCAL_DOWNLOAD_TARGET_FILENAME}`;
+    if (fs.existsSync(LOCAL_BINARY_PATH)) {
+      return;
+    }
+    await clean();
+    await download(DOWNLOAD_URL, archivePath);
+    await extract(archivePath);
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+async function clean(): Promise<void> {
+  if (fs.existsSync(LOCAL_DOWNLOAD_TARGET_PATH)) {
+    fs.removeSync(LOCAL_DOWNLOAD_TARGET_PATH);
+  }
+  fs.mkdirsSync(LOCAL_DOWNLOAD_TARGET_PATH);
+}
+
+function download(url: string, localTargetPath: string): Promise<void> {
```

Comment:
This has been largely extracted from [tracecompass-server.ts](https://github.com/eclipse-cdt-cloud/cdt-cloud-blueprint/blob/master/scripts/tracecompass-server.ts). It would be nice if that script would invoke the functions from this script then, to avoid duplicating the code.